### PR TITLE
Hotfix/remove patch

### PR DESCRIPTION
**Pull request by waldemarnt**
We are removing the PATCH support due the issue https://github.com/agco/harvesterjs/issues/203 .
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/agco/harvesterjs/pull/206%23issuecomment-289877880%22%2C%20%22https%3A//github.com/agco/harvesterjs/pull/206%23issuecomment-289879745%22%2C%20%22https%3A//github.com/agco/harvesterjs/pull/206%23discussion_r108524258%22%2C%20%22https%3A//github.com/agco/harvesterjs/pull/206%23discussion_r108525153%22%2C%20%22https%3A//github.com/agco/harvesterjs/pull/206%23discussion_r108525500%22%2C%20%22https%3A//github.com/agco/harvesterjs/pull/206%23issuecomment-289921562%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/agco/harvesterjs/pull/206%23issuecomment-289877880%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%5Cn%5B%21%5BCoverage%20Status%5D%28https%3A//coveralls.io/builds/10815734/badge%29%5D%28https%3A//coveralls.io/builds/10815734%29%5Cn%5CnCoverage%20decreased%20%28-1.5%25%29%20to%2082.699%25%20when%20pulling%20%2A%2Ab4d257e26666cab40b5d793a9d7ce332d5a52854%20on%20hotfix/remove-patch%2A%2A%20into%20%2A%2A6bd13dfa81c645dda12262791c4638f356fa6cab%20on%20develop%2A%2A.%5Cn%22%2C%20%22created_at%22%3A%20%222017-03-28T19%3A25%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/2354108%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/coveralls%22%7D%7D%2C%20%7B%22body%22%3A%20%22%5Cn%5B%21%5BCoverage%20Status%5D%28https%3A//coveralls.io/builds/10815852/badge%29%5D%28https%3A//coveralls.io/builds/10815852%29%5Cn%5CnCoverage%20decreased%20%28-1.5%25%29%20to%2082.699%25%20when%20pulling%20%2A%2Ab4d257e26666cab40b5d793a9d7ce332d5a52854%20on%20hotfix/remove-patch%2A%2A%20into%20%2A%2A6bd13dfa81c645dda12262791c4638f356fa6cab%20on%20develop%2A%2A.%5Cn%22%2C%20%22created_at%22%3A%20%222017-03-28T19%3A32%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/2354108%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/coveralls%22%7D%7D%2C%20%7B%22body%22%3A%20%22LGTM%22%2C%20%22created_at%22%3A%20%222017-03-28T22%3A15%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/372075%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ssebro%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b4d257e26666cab40b5d793a9d7ce332d5a52854%20test/includes.spec.js%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/agco/harvesterjs/pull/206%23discussion_r108524258%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Could%20you%20explain%20this%20change%3F%20It%20seems%20you%20followed%20an%20TODO%20comment%20that%20led%20you%20to%20this%20change.%22%2C%20%22created_at%22%3A%20%222017-03-28T20%3A08%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/10089668%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lucaslago%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20had%20to%20rewrite%20tests%20on%20which%20PATCH%20were%20used%20to%20use%20PUT%20instead%22%2C%20%22created_at%22%3A%20%222017-03-28T20%3A13%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/5835706%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/waldemarnt%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/includes.spec.js%3AL13-49%22%7D%2C%20%22Pull%20b4d257e26666cab40b5d793a9d7ce332d5a52854%20test/includes.spec.js%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/agco/harvesterjs/pull/206%23discussion_r108525500%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Didn%27t%20change%20the%20test%20behaviour%2C%20it%27s%20passing%2C%20I%20just%20changed%20it%20to%20be%20an%20object%20as%20the%20PUT%20method%20expects.%22%2C%20%22created_at%22%3A%20%222017-03-28T20%3A14%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/5835706%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/waldemarnt%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/includes.spec.js%3AL13-49%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/agco/harvesterjs/pull/206#issuecomment-289877880'>General Comment</a></b>
- <a href='https://github.com/coveralls'></a> [Coverage Status](https://coveralls.io/builds/10815734)
Coverage decreased (-1.5%) to 82.699% when pulling **b4d257e26666cab40b5d793a9d7ce332d5a52854 on hotfix/remove-patch** into **6bd13dfa81c645dda12262791c4638f356fa6cab on develop**.
- <a href='https://github.com/coveralls'></a> [Coverage Status](https://coveralls.io/builds/10815852)
Coverage decreased (-1.5%) to 82.699% when pulling **b4d257e26666cab40b5d793a9d7ce332d5a52854 on hotfix/remove-patch** into **6bd13dfa81c645dda12262791c4638f356fa6cab on develop**.
- <a href='https://github.com/ssebro'></a> LGTM
- [ ] <a href='#crh-comment-Pull b4d257e26666cab40b5d793a9d7ce332d5a52854 test/includes.spec.js 6'></a> &nbsp;<b><a href='https://github.com/agco/harvesterjs/pull/206#discussion_r108524258'>File: test/includes.spec.js:L13-49</a></b>
- <a href='https://github.com/lucaslago'></a> Could you explain this change? It seems you followed an TODO comment that led you to this change.
- <a href='https://github.com/waldemarnt'></a> I had to rewrite tests on which PATCH were used to use PUT instead
- [ ] <a href='#crh-comment-Pull b4d257e26666cab40b5d793a9d7ce332d5a52854 test/includes.spec.js 14'></a> &nbsp;<b><a href='https://github.com/agco/harvesterjs/pull/206#discussion_r108525500'>File: test/includes.spec.js:L13-49</a></b>
- <a href='https://github.com/waldemarnt'></a> Didn't change the test behaviour, it's passing, I just changed it to be an object as the PUT method expects.

<a href='https://www.codereviewhub.com/agco/harvesterjs/pull/206?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/agco/harvesterjs/pull/206?approve=1'></a>&nbsp;<a href='https://github.com/agco/harvesterjs/pull/206'></a>
<a href='#crh-end'></a>

---
This change is [Reviewable](https://reviewable.io/reviews/agco/harvesterjs/206)

***
***
***
***
***
***

### CODE REVIEW

**Comment by ssebro** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/agco/harvesterjs/pull/206'>Refresh</a>.<a href='#crh-update-none'></a>

File: test/includes.spec.js [hunk 1]
```diff
@@ -13,30 +13,36 @@ describe("includes", function () {
     function setupLinks(_ids) {
         ids = _ids;
 
-        function link(url, path, value) {
+        function link(resource, resourceId, link, linkId) {
+          const url = `/${resource}/${resourceId}`;
```
Comment (lucaslago) [inline test/includes.spec.js]: Could you explain this change? It seems you followed an TODO comment that led you to this change.

File: test/includes.spec.js [hunk 2]
```diff
@@ -13,30 +13,36 @@ describe("includes", function () {
...
+
             return new Promise(function (resolve) {
-                var data = [
-                    {
-                        op: 'replace',
-                        path: path,
-                        value: value
+              var data = {
```
Comment (lucaslago) [inline test/includes.spec.js]: data is an object now, is this intentional? What's the reason behind this contract change?

File: test/includes.spec.js (hunk 1 above)
Comment (waldemarnt) [inline test/includes.spec.js]: I had to rewrite tests on which PATCH were used to use PUT instead

File: test/includes.spec.js (hunk 2 above)
Comment (waldemarnt) [inline test/includes.spec.js]: Didn't change the test behaviour, it's passing, I just changed it to be an object as the PUT method expects.

**Comment by ssebro** [comment]
LGTM

**Pull request merged**